docs/api.yaml

```diff
@@ -0,0 +1,925 @@
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/subscription'
+        default:
+          $ref: '#/components/responses/error_response'
+
+    post:
+      description: >-
+        Create a new subscription under the given tenant.
+      tags: 
+        - Subscription
+      requestBody:
+        description: >-
+          Subscription name will be used from path parameter. 
+          #TODO: define a different schema for it.
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/subscription'
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/subscription'
+        default:
+          $ref: '#/components/responses/error_response'
+          
+    put:
+      description: >-
+        Update the mutable proerties of the given subscription.
+      tags: 
+        - Subscription
+      requestBody:
+        description: >-
+          Subscription name will be used from path parameter. 
+          #TODO: define a different schema for it.
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/subscription'
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/subscription'
+        default:
+          $ref: '#/components/responses/error_response'
+          
+    delete:      
+      description: >-
+        Delete the specified subscription.
+      tags: 
+        - Subscription
+      responses:
+        '200':
+          description: OK
+        default:
+          $ref: '#/components/responses/error_response'
+  
+  
+  /v1/tenants/{tenant_name}/subscriptions/{sub_name}/state:
+    parameters:
+      - $ref: '#/components/parameters/tenanat_param'
+      - $ref: '#/components/parameters/sub_param'
+      
+    get:
+      description: >-
+       Get the state of the subscription.
+      tags: 
+        - Subscription
+      responses:
+        '200':
+          $ref: '#/components/responses/state_response'
+
+        default:
+          $ref: '#/components/responses/error_response'
+    put:
+      description: >-
+       Perform the specified operation asynchronously to update the Subscription state. 
+       Allowed operations are block, unblock, start and stop.
+      tags: 
+        - Subscription
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/state_request'
+      responses:
+        '200':
+          description: OK
+        default:
+          $ref: '#/components/responses/error_response'
+  
+          
+  /v1/tenants/{tenant_name}/subscriptions/{sub_name}/status:
```

Comment:
how is this status different from state.

This state vs status, led me to search the difference between them. I think I align with this https://softwareengineering.stackexchange.com/questions/219351/state-or-status-when-should-a-variable-name-contain-the-word-state-and-w

I can understand the active, inactive (/ blocked), running, stopped. being the status of the subscription. Similarly, active, inactive (/blocked) for the topic status.

What is consumer_status exactly. Is this for internal details of the consumer, lag and other stuff? Internal Stats?

Reply:
intresting.. I was thinking of it as "state" being more like state-machine state, status being more detailed running info. kind of single value and multi-value attribute.
Both info as of now intended for external consumption.